interpreter.js

```diff
@@ -1945,6 +1945,23 @@ Interpreter.prototype.nativeToPseudo = function(nativeObj) {
       nativeObj instanceof RegExp) {
     return this.createPrimitive(nativeObj);
   }
+
+  if (nativeObj instanceof Function) {
+    var wrapper = (function(interpreter, nativeObj){
+      return function() {
+        return interpreter.nativeToPseudo(
+          nativeObj.apply(interpreter,
+            Array.prototype.slice.call(arguments)
+              .map(function(i) {
+                return interpreter.pseudoToNative(i);
+              })
+          )
+        );
+      };
+    })(this, nativeObj);
```

Comment:
Out of curiosity, why this:
```
var wrapper = (function(interpreter, nativeObj){
  return function() {
    ...
  };
})(this, nativeObj);
```
instead of this:
```
var interpreter = this;
var wrapper = function() {
  ...
};
```
It seems like there's an unneeded scope being created here.